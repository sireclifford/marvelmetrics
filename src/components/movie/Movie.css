.movie-item {
  display: flex;
  justify-content: space-between;
  padding: 10px;
}

.movie-title {
  display: flex;
  flex-direction: column;
  justify-content: center;
  flex: 2 2 auto;
}

.movie-title h2,
.movie-title p {
  margin: 0;
}

.movie-title p {
  font-size: 0.8rem;
}

.movie-title h2 {
  font-size: 1rem;
}

.movie-details p {
  margin: 0;
}

.seeds-value {
  font-size: 1.2rem;
}

.seeds {
  font-size: 0.8em;
}

.movie-list:nth-child(even) {
  background: #b13867;
}
