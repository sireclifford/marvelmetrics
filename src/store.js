import { configureStore } from '@reduxjs/toolkit';
import MovieSlice from './redux/MovieSlice';

export default configureStore({
  reducer: {
    movies: MovieSlice,
  },
});
