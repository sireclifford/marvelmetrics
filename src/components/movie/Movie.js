/* eslint-disable react/prop-types */
/* eslint-disable camelcase */
import React from 'react';
import { useParams } from 'react-router-dom';
import './Movie.css';

const Movie = (props) => {
  const { movie_id } = useParams();
  const movie = props;
  const {
    movie: { title, rating },
  } = props;
  let content;

  if (Object.keys(movie).length === 0) {
    content = <div>No data found</div>;
  } else {
    content = (
      <div className="movie-item">
        <div className="movie-title">
          <h2>{title}</h2>
          <p>
            Genre:
            {movie_id}
          </p>
        </div>
        <div className="movie-details">
          <p className="seeds-value">{rating}</p>
          <p className="seeds">Rating</p>
        </div>
      </div>
    );
  }

  return <div className="movie-list">{content}</div>;
};

export default Movie;
