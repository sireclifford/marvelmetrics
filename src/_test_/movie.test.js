import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../store';
import Movie from '../components/movie/Movie';

it('Test renders with data', () => {
  const movie = {
    movie_id: '1',
    title: 'test',
    rating: '1',
  };
  const tree = renderer.create(
    <Provider store={store}>
      <Movie movie={movie} />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

it('Test renders with no data', () => {
  const movie = {};
  const tree = renderer.create(
    <Provider store={store}>
      <Movie movie={movie} />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
