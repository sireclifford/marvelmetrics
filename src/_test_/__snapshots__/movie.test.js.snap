// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test renders with data 1`] = `
<div
  className="movie-list"
>
  <div
    className="movie-item"
  >
    <div
      className="movie-title"
    >
      <h2>
        test
      </h2>
      <p>
        Genre:
      </p>
    </div>
    <div
      className="movie-details"
    >
      <p
        className="seeds-value"
      >
        1
      </p>
      <p
        className="seeds"
      >
        Rating
      </p>
    </div>
  </div>
</div>
`;

exports[`Test renders with no data 1`] = `
<div
  className="movie-list"
>
  <div
    className="movie-item"
  >
    <div
      className="movie-title"
    >
      <h2 />
      <p>
        Genre:
      </p>
    </div>
    <div
      className="movie-details"
    >
      <p
        className="seeds-value"
      />
      <p
        className="seeds"
      >
        Rating
      </p>
    </div>
  </div>
</div>
`;
